github/resource_github_repository_webhook.go

```diff
@@ -46,8 +49,38 @@ func resourceGithubRepositoryWebhook() *schema.Resource {
 				Set:      schema.HashString,
 			},
 			"configuration": {
-				Type:     schema.TypeMap,
+				Type:     schema.TypeList,
+				MaxItems: 1,
 				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"url": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"content_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"secret": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+							DiffSuppressFunc: func(k, oldV, newV string, d *schema.ResourceData) bool {
+								maskedSecret := "********"
```

Comment:
I haven't really worked with sensitive attributes, but it feels strange to me for a maskedSecret to ALWAYS be 8 asterisks, is this code robust?

Reply:
GitHub unfortunately doesn't document this, but I assumed it's a static placeholder they just return. Why would/should the API ever return different number of asterisks?

btw. this is unrelated to how we treat sensitive data in the schema via `Sensitive`, it's more related to how the backend API (GitHub in this case) implements it.

Reply:
okay, I think a code comment might be appropriate in this case mentioning it appears to be an undocumented GH api response.